src/cluster.c

```diff
@@ -4899,10 +4896,7 @@ int clusterDelSlot(int slot) {
     /* Cleanup the channels in master/replica as part of slot deletion. */
     list *nodes_for_slot = clusterGetNodesInMyShard(n);
     serverAssert(nodes_for_slot != NULL);
-    listNode *ln = listSearchKey(nodes_for_slot, myself);
-    if (ln != NULL) {
-        removeChannelsInSlot(slot);
-    }
+    listSearchDel(nodes_for_slot, myself);
```

Comment:
This doesn't seem right. `removeChannelsInSlot` does some completely different things.